toolkit-test-app/src/main/java/com/esri/arcgisruntime/toolkit/test/popup/PopupViewTestActivity.kt

```diff
@@ -0,0 +1,172 @@
+/*
+ * Copyright 2021 Esri
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.esri.arcgisruntime.toolkit.test.popup
+
+import android.os.Bundle
+import android.util.Log
+import android.view.MotionEvent
+import android.view.View
+import android.widget.Toast
+import androidx.activity.viewModels
+import androidx.appcompat.app.AppCompatActivity
+import androidx.databinding.DataBindingUtil
+import com.esri.arcgisruntime.data.Feature
+import com.esri.arcgisruntime.geometry.GeometryType
+import com.esri.arcgisruntime.layers.FeatureLayer
+import com.esri.arcgisruntime.mapping.ArcGISMap
+import com.esri.arcgisruntime.mapping.view.DefaultMapViewOnTouchListener
+import com.esri.arcgisruntime.mapping.view.MapView
+import com.esri.arcgisruntime.portal.Portal
+import com.esri.arcgisruntime.portal.PortalItem
+import com.esri.arcgisruntime.toolkit.extension.logTag
+import com.esri.arcgisruntime.toolkit.popup.PopupView
+import com.esri.arcgisruntime.toolkit.popup.PopupViewModel
+import com.esri.arcgisruntime.toolkit.test.R
+import com.esri.arcgisruntime.toolkit.test.databinding.ActivityPopupBinding
+import com.esri.arcgisruntime.toolkit.util.observeEvent
+import com.google.android.material.bottomsheet.BottomSheetBehavior
+import kotlin.math.roundToInt
+
+/**
+ * Activity to show usages of [PopupView] and [PopupViewModel].
+ */
+class PopupViewTestActivity : AppCompatActivity() {
+
+    private lateinit var bottomSheetBehavior: BottomSheetBehavior<View>
+    private lateinit var mapView: MapView
+    private val popupViewModel: PopupViewModel by viewModels()
+
+    val map: ArcGISMap by lazy {
+        val portal = Portal("https://arcgisruntime.maps.arcgis.com/")
+        val portalItem = PortalItem(portal, "16f1b8ba37b44dc3884afc8d5f454dd2")
+        val map = ArcGISMap(portalItem)
+        map
+    }
+
+    private val identifiableLayer: FeatureLayer?
+        get() {
+            return map.operationalLayers?.filterIsInstance<FeatureLayer>()?.filter {
+                (it.featureTable?.geometryType == GeometryType.POINT)
+                    .and(it.isVisible)
+                    .and(it.isPopupEnabled && it.popupDefinition != null)
+            }?.get(0)
+        }
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+
+        val binding: ActivityPopupBinding =
+            DataBindingUtil.setContentView(this, R.layout.activity_popup)
```

Comment:
Do we need the binding here at all? You could get a reference to the MapView and BottomSheetContainer directly from the layout. If you don't need the binding for anything else, I think you can ommit it.

Reply:
Yes, we do need binding here. 

>You could get a reference to the MapView and BottomSheetContainer directly from the layout 

That would mean we will have to use Kotlin Synthetics(which is a part of Kotlin Android Extensions that have been deprecated).

https://developer.android.com/topic/libraries/view-binding/migration#activity-fragment

Reply:
Good point 👍